include/utils/types.hpp

```diff
@@ -30,8 +30,9 @@ using ModuleConfigs = std::map<std::string, ConfigMap>;
 using Array = json::array_t;
 using Object = json::object_t;
 // TODO (aw): can we pass the handler arguments by const ref?
-using Handler = std::function<void(json)>;
+using Handler = std::function<void(const std::string&, json)>;
 using StringHandler = std::function<void(std::string)>;
+using StringPairHandler = std::function<void(const std::string& topic, const std::string& data)>;
```

Comment:
Not sure, but function signatures usually are not named (using `topic` and `data` here) - although it improves readability.  For this, type aliases might be an option (`using Topic = std::string`).